src/main/groovy/com/bmuschko/gradle/docker/tasks/container/DockerCreateContainer.groovy

```diff
@@ -265,7 +266,7 @@ class DockerCreateContainer extends DockerExistingImage {
 
     void withEnvVar(def key, def value) {
         if (envVars.getOrNull()) {
-            envVars.get().put(key, value)
+            envVars.put(key, value)
```

Comment:
`MapProperty.get()` returns an `ImmutableMap` copy of the property.  Need to call `MapProperty.put()` directly in order to modify the property.